test/shmemx/Makefile.am

```diff
@@ -90,10 +89,14 @@ LDADD += $(top_builddir)/pmi-simple/libpmi_simple.la
 endif
 
 if SHMEMX_TESTS
+
+if HAVE_PTHREADS
 gettid_register_LDFLAGS = $(AM_LDFLAGS) $(PTHREAD_LIBS)
 gettid_register_CFLAGS = $(PTHREAD_CFLAGS)
 gettid_register_LDADD = $(LDADD) $(PTHREAD_CFLAGS)
+endif
 
+if HAVE_CXX
```

Comment:
Same with this `if`.  Is it needed?

Reply:
This `if` is also tangential to this PR, but I think it could probably be removed as well.  Remove it?